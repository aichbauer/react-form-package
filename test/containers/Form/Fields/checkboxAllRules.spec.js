import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';

import {
  Form,
  Field,
  Button,
} from '../../../../src';
import { checkboxAllRules } from '../../../fixtures';

configure({ adapter: new Adapter() });

test('Render Form with checkbox | checkbox all rules', () => {
  const myComponent = (
    <Form>
      <div>
        <div>checkbox</div>
        <Field type="checkbox" id="checkbox" required />
      </div>
      <div>
        <Button id="submit" type="submit" onClick={(state) => state}>submit</Button>
      </div>
    </Form>
  );

  const tree = renderer.create(myComponent).toJSON();
  const wrapper = mount(myComponent);
  const state = wrapper.state();

  expect(state).toEqual(checkboxAllRules);
  expect(tree).toMatchSnapshot();
});
