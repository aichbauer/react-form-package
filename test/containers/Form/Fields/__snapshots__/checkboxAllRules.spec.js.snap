// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Form with checkbox | checkbox all rules 1`] = `
<form>
  <div>
    <div>
      checkbox
    </div>
    <input
      checked={false}
      className="rfp-checkbox"
      id="checkbox"
      invalid={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={true}
      type="checkbox"
      valid={false}
      value={false}
    />
  </div>
  <div>
    <button
      className="rfp-button"
      disabled={false}
      id="submit"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      type="submit"
    >
      submit
    </button>
  </div>
</form>
`;
